File: Exercicios_diarios/Revisão_intermediaria/Sequência Zig Zag.cpp
#include <bits/stdc++.h>
using namespace std;

void findZigZagSequence(vector<int> vetor, int len)
{
    int meio = (len + 1) / 2;
    int esquerda = meio + 1;
    int direita = len - 2;

    sort(vetor.begin(), vetor.end());
    swap(vetor[--meio], vetor[len - 1]);

    while (esquerda <= direita)
    {
        swap(vetor[esquerda], vetor[direita]);
        esquerda += 1;
        direita -= 1;
    }

    for (int i = 0; i < len; i++)
    {
        if (i > 0)
            cout << " ";
        cout << vetor[i];
    }
    cout << endl;

}

int main()
{
    int len, num, cont_casos;
    cin >> cont_casos;

    for (int i = 1; i <= cont_casos; i++)
    {
        cin >> len;
        vector<int> vetor;

        for (int i = 0; i < len; i++)
        {
            cin >> num;
            vetor.push_back(num);
        }
        findZigZagSequence(vetor, len);
    }

    return 0;
}

File: Python/Exercicios_diarios/Revisão_intermediaria/Sequência Zig Zag.py
def findZigZagSequence(vetor, len):
    vetor.sort()
    meio = int((len - 1)/2)
    vetor[meio], vetor[len-1] = vetor[len-1], vetor[meio]

    esquerda = meio + 1
    direita = len - 2
    while(esquerda <= direita):
        vetor[esquerda], vetor[direita] = vetor[direita], vetor[esquerda]
        esquerda = esquerda + 1
        direita = direita - 1

    for i in range(len):
        if i == len-1:
            print(vetor[i])
        else:
            print(vetor[i], end=' ')
    return 0


test_cases = int(input())
for cs in range(test_cases):
    len = int(input())
    vetor = list(map(int, input().split()))
    findZigZagSequence(vetor, len)
